torch/csrc/jit/codegen/cuda/lower_predicate.cpp

```diff
@@ -0,0 +1,208 @@
+    for (auto expr : exprs_copy) {
+      handle(expr);
+    }
+
+    for_loops_structure_.pop_back();
+  }
+
+  void handle(kir::IfThenElse* ite) {
+    TORCH_INTERNAL_ASSERT(ite->cond() != nullptr);
+
+    // If ite already has Bool conditional, handle internal expressions
+    // Otherwise, generate conditional and update predicate
+    if (ite->cond()->hasValue()) {
+      const auto then_exprs_copy = ite->thenBody().exprs();
+      for (auto expr : then_exprs_copy) {
+        handle(expr);
+      }
+
+      const auto else_exprs_copy = ite->elseBody().exprs();
+      for (auto expr : else_exprs_copy) {
+        handle(expr);
+      }
+    } else {
+      auto conditional = generateConditional(ite->cond(), ite);
+      TORCH_INTERNAL_ASSERT(conditional != nullptr);
+      TORCH_INTERNAL_ASSERT(conditional->isA<kir::Bool>());
+
+      // If predicate is not required
+      if (conditional->isConst() && conditional->value().value()) {
+        TORCH_INTERNAL_ASSERT(ite->elseBody().empty());
+
+        // Handle internal expressions
+        const auto then_exprs_copy = ite->thenBody().exprs();
+        for (auto expr : then_exprs_copy) {
+          handle(expr);
+        }
+
+        if (for_loops_structure_.empty()) {
+          TORCH_INTERNAL_ASSERT(then_exprs_copy.size() == 1);
+          auto expr = then_exprs_copy.front();
+          updateTensorViewDefinitionScope(expr, nullptr);
+          expr_replacement_map_.insert({ite, expr});
+        } else {
+          // Add all expressions in Then body to current for-loop
+          for (auto expr : then_exprs_copy) {
+            for_loops_structure_.back()->body().insert_before(ite, expr);
+            updateTensorViewDefinitionScope(
+                expr, &for_loops_structure_.back()->body());
+          }
+          // Remove the unnecessary IfThenElse node
+          for_loops_structure_.back()->body().erase(ite);
+        }
+      } else {
+        // Update bool conditional in-place
+        ite->cond()->setValue(conditional);
+        handle(ite);
+        TORCH_INTERNAL_ASSERT(ite->cond()->value() != nullptr);
+      }
+    }
+  }
+
+  // The FusionOmitPredicate tests detect if a TensorView has a
+  // predicate by checking the parent scope of the lowered
+  // kir::TensorView. All kir::TensorViews are guarded by an
+  // intermediate predicate, while the boolean conditional is generated
+  // on the TensorIndex during this lowering pass. This function updates
+  // the kir::TensorView, so the check runs correctly.
+  void updateTensorViewDefinitionScope(kir::Expr* expr, kir::Scope* scope) {
+    if (!expr->outputs().empty()) {
+      auto out = expr->outputs().front();
+      if (out->isA<kir::TensorIndex>()) {
+        auto out_ti = out->as<kir::TensorIndex>();
+        out_ti->view()->definition()->setScope(scope);
+      }
+    }
+  }
+
+  // Generate conditional according to PredicateType
+  kir::Bool* generateConditional(kir::Predicate* pred, kir::IfThenElse* ite) {
```

Comment:
Please comment what `ite` represents here. It seems it's just `nullptr` sometimes.

Reply:
The `ite` argument is only used to get the `unrolled_loop` for the `UnswitchPredicate`. I'll refactor this function to remove the argument.